### DISCUSSION TITLE
New jnode site on GitHub Pages?

### DISCUSSION THREAD

**Discussion by ghost** [jnode-revisited/jnode#2] [Q&A] [answered]
Hi.

#### concept
I would like to contribute by creating a new site for jnode. *What do you all think of the idea of creating a new site for jnode?*

#### why?
1. jnode is in the process of revision, or new revision. So... I believe the jnode website would need a redesign or revision, or new revision too
2. I'm thinking of adding jnode to GitHub Pages hosting because GitHub Pages has a free plan to host an open source website. An alternative to very used GitHub Pages is Netlify or Vercel. 

#### How do you intend to do this?
My general idea is to have 3 websites. The first site is for general community information, or general links. The second site is the technical documentation. The third site is for showcase, where we demonstrate applications made in jnode and other use cases. 

#### But... How do you intend to do this?
1. The main site: I think of using only html,css,js to create the main page in GitHub Pages.
2. I'm thinking of using the [docsify.js](https://github.com/docsifyjs/docsify) framework to manage and maintain a technical documentation site for jnode. This documentation will live on Vercel.
3. In the case of jnode's third "showcase" site, I'm thinking of using hugo. This site serves only to index information of applications written in jnode. In this use case, consider using Netlify.

#### What do you expect to have 3 different hosting locations on each page?
1. The first reason is traffic, you would expect the technical documentation pages to have a different amount of traffic or hits than the main or presentation page.  In all hosting there are limits on the amount of traffic available, I hope not to exceed this limit I have 3 different hosting. Generally these traffic limits are limited by account type in each of the hosting, something like: "free plan", "premium" or "company" with x limit of traffic. *I initially thought about this, as I want to adhere to the term of service with each hosting on the free plan.*
2. Another good reason is that each page has a different target audience, usually programmers, developers or coders who read the technical documentation. Those who access the main page are people interested in jnode, who can be users, programmers, developers or coders or not. And usually whoever accesses the showcase page is more interested if there are any apps they can use or like on jnode.

**Comment by tripleo1** [comment, marked as answer]
I think great.

Also, if you know how, you can create a website on Github Pages.  I am happy to host it and add you as a maintainer or whatever so you get full access.

**Reply by tripleo1** [reply]
I forgot to add, what about all the documentation that exists on `jnode.org`?

**Reply by ghost** [reply]
Hi tripleo1.

> I think great.
> Also, if you know how, you can create a website on Github Pages. I am happy to host it and add you as a maintainer or whatever so you get full access.

thank you for feedback.

> I forgot to add, what about all the documentation that exists on jnode.org?

I'm thinking of using the docsify.js framework to manage and maintain a technical documentation site for jnode.  This documentation will live on Vercel. **Note:** I don't know why when I try to access the technical documentation page it displays this warning. Please, see this: https://www.jnode.org/node/132

![image](https://user-images.githubusercontent.com/111701513/208269708-599d5110-94e4-494b-bb4d-fbdbcd761285.png)

**Comment by tripleo1** [comment]
> docsify.js 

looks cute. have at it!

> web server is down

Either somebody didn't pay their bill or CloudFlare sucks.  Could be either.
See https://web.archive.org/web/20190718173028/https://www.jnode.org/node/132

> 3 websites

My only suggestion is to maintain them with backups, for example hosting the source on Github or something like it.

----

If you need anything let me know.

You can reach me here. (@tripleo1 )

**Reply by ghost** [reply]
Hi tripleo1.

> My only suggestion is to maintain them with backups, for example hosting the source on Github or something like it.

thank you for feedback.

**Reply by tripleo1** [reply]
Look here:

https://github.com/marketplace?type=&verification=&query=documentation+

and here:

https://github.com/marketplace/actions/slate-documentation-builder

https://github.com/slatedocs/slate

https://slatedocs.github.io/slate/#introduction

I have no connection to slate, just found out about it (personally, I use read the docs), but it looks nice.

**Reply by tripleo1** [reply]
Not slate, but:

https://github.com/jnode-revisited/jnode-revisited.github.io

You're welcome to approve your invite and get started as you please.

**Reply by ghost** [reply]
Hi tripleo1.

> Look here:
> https://github.com/marketplace?type=&verification=&query=documentation+
> and here:
> https://github.com/marketplace/actions/slate-documentation-builder
> https://github.com/slatedocs/slate
> https://slatedocs.github.io/slate/#introduction

Thank you for feedback.

> Not slate, but:
> https://github.com/jnode-revisited/jnode-revisited.github.io
> You're welcome to approve your invite and get started as you please.

Thank you for feeback... but... I would like to develop things independently. So... *wouldn't it be possible for you to create a hello world repository and I send the pull request?*

**Comment by tripleo1** [comment]
https://jnode-revisited.github.io/

https://github.com/jnode-revisited/jnode-revisited.github.io/blob/main/README.md

Please note that the site is set up for Jekyll, that's why I made it open for your own design. I didn't want to make it as if you were following me -- it's your show! 

Any changes you want, create an issue on that repo.

Thx.

**Reply by tripleo1** [reply]
My advice is to accept the invite, throw some stuff up there and we can fix it up later.

Release early and often!

**Reply by tripleo1** [reply]
Oh, ok. I see what you mean. Yeah do PRs.

**Reply by ghost** [reply]
Hi tripleo1.

> My advice is to accept the invite, throw some stuff up there and we can fix it up later.
> Release early and often!

Thanks for the feedback, but I find it easier to just send PR yet.

> Oh, ok. I see what you mean. Yeah do PRs.

Yes... but... *Is this site for the technical documentation or will it be the main site? (just confirming what will be done.)*

> https://jnode-revisited.github.io/
> https://github.com/jnode-revisited/jnode-revisited.github.io/blob/main/README.md
> Please note that the site is set up for Jekyll, that's why I made it open for your own design. I didn't want to make it as if you were following me -- it's your show!
> Any changes you want, create an issue on that repo.
> Thx.

thank you for feedback again.

**Comment by ghost** [comment]
Hi all, hi @tripleo1.

#### 1. Idea & prototype 1 *"landing page" or "main page"*
Idea & prototype for "jnode-revisited.github.io": *Jnode-revisited+Jnode.org+Github-Pages+Theme HTML, CSS, JS from Alamehan+GitHub Discussions+Docsify+Vercel+Netlify+Hugo ^^*
- url: `jnode-revisited.netlify.app -> homepage`

<details>
  <summary>1.1 img</summary>
  
![image](https://user-images.githubusercontent.com/111701513/208491977-2ba1e864-ec5a-4bf2-b708-c7c88918c455.png)

**image-description:** This website was redesigned and inspired by the open source project [alamehan](https://alamehan.github.io/), source-code [github/alamehan/alamehan.github.io](https://github.com/alamehan/alamehan.github.io)

</details>

#### 2. Idea & prototype 2 *"docs"*
- url: `jnode-revisited.netlify.app/docs/# -> documentation`

<details>
  <summary>2.1 img-before</summary>
  
![image](https://user-images.githubusercontent.com/111701513/208280222-0bd8b6cd-9aa2-430e-8087-2aab79b9cc9f.png)

</details>

<details>
  <summary>2.2 img-after</summary>
  
![image](https://user-images.githubusercontent.com/111701513/208537473-740b8ecc-9e06-45ef-8666-9fd68f28c896.png)

**image-description:** This website was redesigned and inspired by the open source project [jhildenbiddle.github.io/docsify-themeable](https://jhildenbiddle.github.io/docsify-themeable/#/), source-code [github/jhildenbiddle/docsify-themeable](https://github.com/jhildenbiddle/docsify-themeable)


</details>

#### 3. question/feedback
*What do you all think of the idea?*

**Reply by tripleo1** [reply]
Nice start. Send a PR when you're ready.

**Reply by ghost** [reply]
Hi tripleo1.

> Nice start. Send a PR when you're ready.

thank you for feedback. Please, see my [PR](https://github.com/jnode-revisited/jnode-revisited.github.io/pull/8) or my fork [github/codehangen/jnode-revisited.github.io](https://github.com/codehangen/jnode-revisited.github.io)
